example/src/main/java/com/ryanharter/auto/value/gson/example/UnrecognisedExample.java

```diff
@@ -0,0 +1,36 @@
+// modified by mapbox
+package com.ryanharter.auto.value.gson.example;
+
+import com.google.auto.value.AutoValue;
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.ryanharter.auto.value.gson.UnrecognisedJsonProperties;
+
+import java.util.Map;
+
+@AutoValue
+public abstract class UnrecognisedExample {
+
+  public abstract int knownField();
+
+  public static UnrecognisedExample.Builder builder() {
+    return new AutoValue_UnrecognisedExample.Builder();
+  }
+
+  public static TypeAdapter<UnrecognisedExample> typeAdapter(Gson gson) {
+    return new AutoValue_UnrecognisedExample.GsonTypeAdapter(gson);
+  }
+
+  @Nullable
+  @UnrecognisedJsonProperties
+  abstract Map<String, Object> unknownProperties();
+
+  @AutoValue.Builder
+  public static abstract class Builder {
+    public abstract Builder knownField(int value);
+    abstract Builder unknownProperties(@Nullable Map<String, Object> value);
```

Comment:
Should this method also have its annotation so that it should be used to insert the unrecognized fields? Otherwise the creator of the class has to know that the method from the class has to have the exact same name as the method from the builder. That's fine as well but we might want to assert it in the code generator or clearly document in the annotation docs.

Reply:
> has to have the exact same name as the method from the builder.

AutoValue requires this already, doesn't it?